internal/ingress/controller/config/config.go

```diff
@@ -882,9 +885,9 @@ func NewDefault() Configuration {
 		ComputeFullForwardedFor:          false,
 		ProxyAddOriginalURIHeader:        false,
 		GenerateRequestID:                true,
-		HTTP2MaxFieldSize:                "4k",
-		HTTP2MaxHeaderSize:               "16k",
-		HTTP2MaxRequests:                 1000,
+		HTTP2MaxFieldSize:                "",
```

Comment:
@rikatz @strongjz do you see any unforeseen side affects to setting the defaults to these as their nil values?

Reply:
We don't know who is using it, and changing defaults could break things. @tao12345666333 @rikatz might think differently tho.

Reply:
@strongjz My thinking on this would be if they haven't explicitly set these then they wont be used (as the nginx.tmpl wont add them), and as we default the directives that are replaced are in fact explicitly set

- http2_max_field_size, http2_max_header_size -> [large_client_header_buffers](https://github.com/kubernetes/ingress-nginx/blob/main/internal/ingress/controller/config/config.go#L900)
- http2_max_requests -> [keepalive_requests](https://github.com/kubernetes/ingress-nginx/blob/main/internal/ingress/controller/config/config.go#L900)

Reply:
But I also might be misunderstanding

Reply:
sounds good to me, then.